dist/ExtUtils-ParseXS/lib/perlxs.pod

```diff
@@ -1,2065 +1,4706 @@
+I<isn't> recognised by the XS parser as a valid C preprocessor directive,
+is treated as an XS comment line.
+
+It is recommended to include at least one space before an XS C<#> comment
+to avoid any possible confusion with C preprocessor directives.
+
+If an XS comment line ends with a backslash, then the line following it is
+treated as part of that comment line and is also discarded.
+
+=back
+
+Once that basic textual preprocessing has been performed, the main XS
+parsing takes place. XS syntax is very line-orientated. XS lines and
+sections mostly start with a keyword of the form;
+
+    /^\s*[A-Z_]+:/
+
+It is best to position file-scoped keywords at column one, while
+XSUB-scoped keywords are best indented. This may avoid surprises with edge
+cases in the XS parser.
+
+Keywords can be either single line, e.g. C<PROTOTYPES: ENABLE>, or
+multi-line. The latter consume lines until the next keyword, or until the
+possible start of a new XSUB (C</\n\n\S/>), or to EOF. Multi-line keywords
+treat the rest of the text on the line which follows the keyword as the
+first line of data. The exception to this is keywords which introduce a
+block of code, such as C<CODE:>  or C<BOOT:>, which silently ignore the
+rest of the first line. (Yes, this is a implementation flaw.)
+
+It is best to include a blank line between each file-scoped item, and
+before the start of each XSUB. While some items I<are> processed correctly
+if they are on the line immediately preceding the start of an XSUB, the
+parser is inconsistent in their handling.
+
+An XSUB ends when C</\n\n\S/> is encountered: i.e. a blank line followed
+by something on column one. (This is why it's recommended to indent
+XSUB-scoped keywords.) If the thing at column one matches any of the items
+which can appear in between XSUBs (such as file-scoped keywords) then it,
+and any subsequent lines, are processed as such. Anything starting on
+column one which isn't otherwise recognised, is interpreted as the first
+line of the next XSUB definition. In particular it is interpreted as the
+return type of the XSUB: this can lead to weird errors when something is
+unexpectedly interpreted as the start of a new XSUB, such as C</* */>,
+which isn't valid in the XS half of the file apart from within code
+blocks.
+
+Some multi-line keywords, such as C<C_ARGS>, are treated as just a single
+uninterpreted multi-line string. Others, such as C<OUTPUT>, have a
+specific per-line syntax, where each line within the section is parsed.
+Finally, code blocks such as C<CODE> are just copied as-is to the output C
+file (possibly sandwiched between C<#line> directives to ensure that
+compiler error messages report from the correct location).
+
+The XS parser doesn't recognise C comments, so don't use them apart from
+in C code (e.g. not in an XSUB signature). More generally, the XS parser
+doesn't understand C syntax or semantics; it just uses crude regexes to
+parse the XS file. For example in an XSUB declaration like
+
```

Comment:
This example doesn't really explain how the declaration is parsed. Does it get it right? Or horribly wrong?